src/block_cache/block_chunkcache.c

```diff
@@ -20,6 +20,40 @@
 #define WT_CHUNK_OFFSET(chunkcache, offset) \
     (wt_off_t)(((size_t)offset / (chunkcache)->chunk_size) * (chunkcache)->chunk_size)
 
+/*
+ * __bitmap_find_free --
+ *     Iterate through the bitmap to find a free chunk in the cache.
+ */
+static int
+__bitmap_find_free(WT_SESSION_IMPL *session, uint8_t *bit_index)
+{
+    WT_CHUNKCACHE *chunkcache;
+    size_t bitmap_size, bit_remainder;
+    uint8_t map_byte;
+
+    chunkcache = &S2C(session)->chunkcache;
+    bitmap_size = chunkcache->capacity / chunkcache->chunk_size;
+
+    for (size_t i = 0; i < bitmap_size; i++) {
+        map_byte = chunkcache->bitmap[i];
+        if (map_byte != 0xff) {
+            for (size_t j = 0; j < 8; j++) {
+                if ((map_byte & (0x01 << j)) == 0) {
+                    *bit_index = ((i * 8) + j);
+                    return (0);
+                }
+            }
+        }
+    }
+    bit_remainder = chunkcache->capacity % chunkcache->chunk_size;
```

Comment:
Yup! That is true. I've just added that so that bitmap_size is the number of bytes - that makes it cleaner. I'll just recalculate bit_remainder based on bitmap_size.

Reply:
done here [0366ff8](https://github.com/wiredtiger/wiredtiger/pull/9442/commits/0366ff84099f6d54731d63f524f8756b77f551e1)